pex/environment.py

```diff
@@ -651,6 +655,20 @@ def _declare_namespace_packages(cls, resolved_dists):
                 )
                 for index, (dist, ns_packages) in enumerate(namespace_packages_by_dist.items())
             )
+            if not pkg_resources:
+                current_interpreter = PythonInterpreter.get()
+                pex_warnings.warn(
+                    "The legacy `pkg_resources` package cannot be imported by the "
+                    "{interpreter} {version} interpreter at {path}.\n"
+                    "These distributions will fail to work properly:\n{dists}".format(
```

Comment:
Should this be an eager error instead? The warning below makes sense—we can salvage the situation but it's not kosher. Here, this will eventually result in an error for the user, right?

Reply:
Also maybe worth mentioning this line:

> These distributions should fix their `install_requires` to include `setuptools

Otherwise the user doesn't know a possible path forward for this issue, unless they already are familiar with this niche problem.

Reply:
An eager error would be bad: we tank the ship for an import that may never happen. When a tool tries to do too much, there is a very maddening side to it and its this.

Your second suggestion is reasonable though.

Reply:
Actually its not really reasonable, since that is not a path forward. The distributions simply won't work at all under 3.12 (unless non-ns-package parts are imported per the point above, in which case moot).